src/orders/components/OrderHistory/ExtendedTimelineEvent.tsx

```diff
@@ -0,0 +1,154 @@
+import { makeStyles, Typography } from "@material-ui/core";
+import Money from "@saleor/components/Money";
+import { TimelineEvent } from "@saleor/components/Timeline";
+import { staffMemberDetailsUrl } from "@saleor/staff/urls";
+import classNames from "classnames";
+import camelCase from "lodash/camelCase";
+import React from "react";
+import { defineMessages, useIntl } from "react-intl";
+
+import { OrderDetails_order_events } from "../../types/OrderDetails";
+
+const useStyles = makeStyles(
+  theme => ({
+    eventSubtitle: {
+      marginBottom: theme.spacing(0.5),
+      marginTop: theme.spacing(1)
+    },
+    header: {
+      fontWeight: 500,
+      marginBottom: theme.spacing(1)
+    },
+    linesTableCell: {
+      paddingRight: theme.spacing(3)
+    },
+    root: { marginTop: theme.spacing(4) },
+    topSpacer: {
+      marginTop: theme.spacing(3)
+    },
+    user: {
+      marginBottom: theme.spacing(1)
+    }
+  }),
+  { name: "OrderHistory" }
+);
+
+export const messages = defineMessages({
+  fulfillmentRefundedProductsTitle: {
+    defaultMessage: "Products refunded",
+    description: "event products list title"
+  },
+  fulfillmentRefundedTitle: {
+    defaultMessage: "Products were refunded by ",
+    description: "event title"
+  },
+  fulfillmentReplacedProductsTitle: {
+    defaultMessage: "Products replaced",
+    description: "event products list title"
+  },
+  fulfillmentReplacedTitle: {
+    defaultMessage: "Products were replaced by ",
+    description: "event title"
+  },
+  fulfillmentReturnedProductsTitle: {
+    defaultMessage: "Products returned",
+    description: "event products list title"
+  },
+  fulfillmentReturnedTitle: {
+    defaultMessage: "Products were returned by",
+    description: "event title"
+  },
+  refundedAmount: {
+    defaultMessage: "Refunded amount",
+    description: "amount title"
+  },
+  refundedShipment: {
+    defaultMessage: "Shipment was refunded",
+    description: "shipment refund title"
+  }
+});
+
+interface ExtendedTimelineEventProps {
+  event: OrderDetails_order_events;
+  orderCurrency: string;
+}
+
+const ExtendedTimelineEvent: React.FC<ExtendedTimelineEventProps> = ({
+  event,
+  orderCurrency
+}) => {
+  const { id, date, type, user, lines, amount, shippingCostsIncluded } = event;
+  const classes = useStyles({});
+  const intl = useIntl();
+
+  const getTitle = (titleType: "Title" | "ProductsTitle") =>
+    messages[`${camelCase(type)}${titleType}`];
```

Comment:
Instead of making this weired selection I would prefer something like this:
```tsx
export const titles = defineMessages({...});
export const productTitles = defineMessages({...});
export const messages = defineMessages({...});
```
and then use it with
```tsx
{intl.formatMessage(titles[type])}
{intl.formatMessage(productTitles[type])}
{intl.formatMessage(messages.refundedAmount)}
{intl.formatMessage(messages.refundedShipment)}
```
I think it will remove unneeded complexity of selecting proper message and make it easier to read for me.